app/src/main/java/to/dev/dev_android/view/main/view/CustomWebViewClient.kt

```diff
@@ -85,4 +91,64 @@ class CustomWebViewClient(
             view?.evaluateJavascript(javascript, null)
         })
     }
+
+    private var networkWatcher: NetworkWatcher? = null
+    private fun registerNetworkWatcher() {
+        unregisterNetworkWatcher()
```

Comment:
Could we replace this line for an `if (networkManager != null) return` statement? I think it would have the same effect.

The `WebViewClient` retries a few times when the request fails (I guess it's default behavior) or something happens that makes this `registerNetworkWatcher()` call multiple times. You can see this happening by adding a breakpoint here.

It's not a problem because it first unregisters an existing watcher (if any). But I think we can avoid  unregistering and registering (which deallocates & allocates the `NetWorkWatcher`) a few times 🙂

Reply:
Alright, so first thing I wanted to say is that the Event is being posted multiple times. This posting of the event multiple times is causing the `registerNetworkWatcher()` to run multiple times. I did the `unregistering` and `registering` because I wanted to make sure that the `coroutineScope` did not change. However, I do think that the coroutine scope will not change.

But yeah, you are correct. I'll do that! :+1: